src/adaptation/icap/ModXact.cc

```diff
@@ -1338,11 +1338,7 @@ void Adaptation::Icap::ModXact::finalizeLogInfo()
     al.adapted_request = adapted_request_;
     HTTPMSGLOCK(al.adapted_request);
 
-    if (adapted_reply_) {
-        al.reply = adapted_reply_;
-        HTTPMSGLOCK(al.reply);
-    } else
-        al.reply = NULL;
+    al.setReply(adapted_reply_);
```

Comment:
Here, we set ICAP ALE::reply way after the corresponding ICAP transaction is over. AFAICT, this is conceptually wrong because that same ALE may be used a lot earlier for various ACL-driven directives. We should be populating ALE fields as we learn ALE information instead. Do you agree?

If yes, then I do _not_ think this PR should fix this problem (because a lot of other fields are already set in this too-late method, and we should probably fix all or most of them at once), but please add an XXX comment to mark this problem. For example:

```suggestion
    // XXX: This reply (and other ALE members!) may have been needed earlier.
    al.setReply(adapted_reply_);
```